### PR TITLE
refine docs about NLB

### PR DESCRIPTION
**Pull request by M00nF1sh**
### Issue
N/A

### Description
refine docs about NLB, mainly
1. organize the documentation of two ways to enable NLB support: loadBalancerClass / `service.beta.kubernetes.io/aws-load-balancer-type` annotation.
2. document the security group rules on worker nodes.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: docs/guide/service/nlb.md [hunk 1]
```diff
@@ -1,74 +1,149 @@
...
+
+=== "With LoadBalancerClass"
+    AWS Load Balancer Controller supports `LoadBalancerClass` feature since [:octicons-tag-24: v2.4.0](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.4.0) release for Kubernetes v1.22+ clusters. 
+     
+    `LoadBalancerClass` feature provides a CloudProvider agnostic way of offloading the reconciliation for Kubernetes Services resources of type `LoadBalancer` to an external controller.
+    
+    When you specify the `spec.loadBalancerClass` to be `service.k8s.aws/nlb` on a Kubernetes Service resource of type `LoadBalancer`, the AWS Load Balancer Controller takes charge of reconciliation by provision an NLB.
+
+    !!! warning
+        - It's not recommended to modify or add the `spec.loadBalancerClass` on an existing Service resource. Instead, delete the existing Service resource and recreate a new one if a change is desired.
```
Comment (kishorj) [inline docs/guide/service/nlb.md:44]: `spec.loadBalancerClass` field is immutable. Users will not run into this situation.

File: docs/guide/service/nlb.md [hunk 2]
```diff
@@ -87,20 +162,40 @@ service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
 ## Subnet tagging requirements
 See [Subnet Discovery](../../deploy/subnet_discovery.md) for details on configuring ELB for public or private placement.
 
-
 ## Security group
-NLB does not currently support managed security groups. For ingress access, the controller adds inbound rules to the node security group for the instance mode, or the ENI security group for the IP mode. In case of multiple
-security groups, the controller expects only one security group tagged with the cluster name as follows:
+AWS currently does not support attach security groups to NLB. To allow inbound traffic from NLB, the controller automatically adds inbound rules to worker node security groups by default.
```
Comment (kishorj) [inline docs/guide/service/nlb.md:166]: `does not suppport attaching security groups` or `does not support security groups`

Comment (kishorj) [inline docs/guide/service/nlb.md:166]: `adds inbound rules to the worker node security groups by default`

File: docs/guide/service/annotations.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ Traffic Routing can be controlled with following annotations:
 
 - <a name="nlb-target-type">`service.beta.kubernetes.io/aws-load-balancer-nlb-target-type`</a> specifies the target type to configure for NLB. You can choose between
 `instance` and `ip`.
-    - `instance` mode will route traffic to all EC2 instances within cluster on the [NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) opened for your service.
+    - `instance` mode will route traffic to all EC2 instances within cluster on the [NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) opened for your service. In this mode, AWS NLB sends traffic to the instances and the kube-proxy on the individual worker nodes then forward it to the Kubernetes pods behind the service.
```
[the remark this answers is not in the document]
Comment (M00nF1sh) [reply docs/guide/service/annotations.md:83]: :D sounds good

File: docs/guide/service/nlb.md [hunk 4]
```diff
@@ -1,74 +1,149 @@
...
-    k8s version 1.22 and later support disabling NodePort allocation by setting the service field `spec.allocateLoadBalancerNodePorts` to `false`. If the NodePort is not allocated for a service port, the controller will fail to reconcile instance mode NLB.
-
-```yaml
-    metadata:
-      name: my-service
-      annotations:
-        service.beta.kubernetes.io/aws-load-balancer-type: "external"
-        service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "instance"
-```
+By default, Kubernetes Service resources of type `LoadBalancer` was reconciled by the Kubernetes controller built into the CloudProvider component of the kube-controller-manager or the cloud-controller-manager(a.k.a. the in-tree controller).
```
[the remark this answers is not in the document]
Comment (M00nF1sh) [reply docs/guide/service/nlb.md:31]: +1

File: docs/guide/service/nlb.md (hunk 1 above)
Comment (M00nF1sh) [reply docs/guide/service/nlb.md:44]: thanks for pointing this out

File: docs/guide/service/nlb.md [hunk 5]
```diff
@@ -1,74 +1,149 @@
...
+
+In order to let AWS Load Balancer Controller manage the reconciliation for Kubernetes Services resources of type `LoadBalancer`, we must offloading the reconciliation from in-tree controller to AWS Load Balancer Controller explicitly.
+
+
+=== "With LoadBalancerClass"
+    AWS Load Balancer Controller supports `LoadBalancerClass` feature since [:octicons-tag-24: v2.4.0](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.4.0) release for Kubernetes v1.22+ clusters. 
+     
+    `LoadBalancerClass` feature provides a CloudProvider agnostic way of offloading the reconciliation for Kubernetes Services resources of type `LoadBalancer` to an external controller.
+    
+    When you specify the `spec.loadBalancerClass` to be `service.k8s.aws/nlb` on a Kubernetes Service resource of type `LoadBalancer`, the AWS Load Balancer Controller takes charge of reconciliation by provision an NLB.
```
[the remark this answers is not in the document]
Comment (M00nF1sh) [reply docs/guide/service/nlb.md:41]: :D

File: docs/guide/service/nlb.md (hunk 2 above)
Comment (M00nF1sh) [reply docs/guide/service/nlb.md:166]: i prefer `does not suppport attaching security groups` to be more specific

Comment (M00nF1sh) [reply docs/guide/service/nlb.md:166]: +1

**Pull request merged**